tracer/span.go

```diff
@@ -121,12 +141,20 @@ func (s *Span) SetMetric(key string, val float64) {
 		return
 	}
 
-	s.mu.Lock()
+	s.Lock()
+	defer s.Unlock()
+
+	// We don't lock spans when flushing, so we could have a data race when
+	// modifying a span as it's being flushed. This protects us against that
+	// race, since spans are marked `finished` before we flush them.
```

Comment:
I think that comment, here, is not really relevant as, indeed, it *IS* locked in that place.